server/src/main/java/org/apache/iotdb/db/utils/datastructure/TVList.java

```diff
@@ -519,7 +518,7 @@ public boolean hasNextTimeValuePair() {
 
       while (cur < size) {
         long time = getTime(cur);
-        if (time < getTimeOffset() || (cur + 1 < size() && (time == getTime(cur + 1)))) {
+        if (isPointDeleted(time) || (cur + 1 < size() && (time == getTime(cur + 1)))) {
```

Comment:
you record the deletion pair in the TVList, when iterating the TVList, it's better to sortAndMerge all deletions.
When calling the Ite, the TVList should be sorted already, you can store the deletions inside the Ite and record and index of current checked deletion. Then you could use a more efficient way to check which point is deleted.

Reply:
Fixed